src/pybind/mgr/orchestrator/module.py

```diff
@@ -703,39 +701,32 @@ def _get_service_specs(self, service_name=None):
         completion = self.list_specs(service_name=service_name)
         self._orchestrator_wait([completion])
         raise_if_exception(completion)
-        specs = completion.result_str()
-        return HandleCommandResult(stdout=specs)
+        specs = completion.result()
+        return HandleCommandResult(stdout=yaml.safe_dump_all(specs))
 
     @_cli_write_command(
         'orch apply',
-        'name=service_type,type=CephChoices,strings=mon|mgr|rbd-mirror|crash|alertmanager|grafana|node-exporter|prometheus '
+        'name=service_type,type=CephChoices,strings=mon|mgr|rbd-mirror|crash|alertmanager|grafana|node-exporter|prometheus,req=false '
         'name=placement,type=CephString,req=false '
         'name=unmanaged,type=CephBool,req=false',
-        'Update the size or placement for a service')
-    def _apply_misc(self, service_type, placement=None, unmanaged=False):
-        placement = PlacementSpec.from_string(placement)
-        placement.validate()
+        'Update the size or placement for a service or apply a large yaml spec')
+    def _apply_misc(self, service_type=None, placement=None, unmanaged=False, inbuf=None):
```

Comment:
imo that should default for unmanaged to `None` and let the ServiceSpec decide which default to take. We may want to change the default for certain Services in the future.

```suggestion
    def _apply_misc(self, service_type=None, placement=None, unmanaged=None, inbuf=None):
```

Reply:
Can we postpone this, till we need it? I'd like to avoid adding a `None` value here.

Reply:
sure

Reply:
sure